### PR TITLE
[Dev todo mvp kotlin] Use more extension functions

### PR DESCRIPTION
**Pull request by GuoTianqi**

### CODE REVIEW

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.kt [hunk 1]
```diff
@@ -51,9 +51,9 @@ class AddEditTaskActivity : AppCompatActivity() {
         var shouldLoadDataFromRepo = true
 
         // Prevent the presenter from loading data from the repository if this is a config change.
-        if (savedInstanceState != null) {
```
Comment (DougSig) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.kt]: `if ( localvar != null)` is arguably more readable than `localvar.let` so please revert back to the if statement.

Comment (GuoTianqi) [reply todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.kt]: I have reverted the commit

**Comment by DougSig** [comment]
Thanks!

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.kt (hunk 1 above)
Comment (arekolek) [reply todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.kt]: @DougSig I thought it was being promoted as a [Kotlin idiom](https://kotlinlang.org/docs/reference/idioms.html#execute-if-not-null):

> Execute if not null:
> ```kotlin
> val data = ...
> 
> data?.let {
>     ... // execute this block if not null
> }
> ```

Comment (DougSig) [reply todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.kt]: ?.<let/apply/also/run> is a language feature, but it comes down to the choice of the programmer and readability of the code. For local vars, using 'if' is safe, and it even makes it easier to have an 'else' clause.

**Pull request merged**